maestro/config/exporter_builder.go

```diff
@@ -21,24 +25,73 @@ type PrometheusExporterConfig struct {
 }
 
 func (p *PrometheusExporterConfig) GetExportersFromMetadata(config types.Metadata, authenticationExtensionName string) (Exporters, string) {
-	endpointCfg := config.GetSubMetadata("exporter")["remote_host"].(string)
-	return Exporters{
+	// Check if exporter metadata is present
+	exporterMetadata := config.GetSubMetadata("exporter")
+	if exporterMetadata == nil {
+		log.Println("exporter metadata is missing")
+		return Exporters{}, ""
+	}
+
+	// Retrieve the remote_host endpoint configuration
+	endpointCfg, ok := exporterMetadata["remote_host"].(string)
+	if !ok {
+		log.Printf("remote_host metadata is missing or not a string %v", exporterMetadata)
+		return Exporters{}, ""
+	}
+
+	// Create PrometheusRemoteWriteExporterConfig with endpoint and authentication
+	exporters := Exporters{
 		PrometheusRemoteWrite: &PrometheusRemoteWriteExporterConfig{
 			Endpoint: endpointCfg,
 			Auth:     Auth{Authenticator: authenticationExtensionName},
 		},
-	}, "prometheusremotewrite"
+	}
+
+	// Check to add X-Scope-OrgID header
+	headersMetadata := config.GetSubMetadata("headers")
+	if headersMetadata != nil {
+		if headerStr, ok := headersMetadata["X-Scope-OrgID"].(string); ok && headerStr != "" {
+			exporters.PrometheusRemoteWrite.Headers = make(map[string]string)
+			exporters.PrometheusRemoteWrite.Headers["X-Scope-OrgID"] = headerStr
+		}
+	}
```

Comment:
It will be better to not have the restriction to only this header, instead we can block the header we don't want.
And also, this is not the place to block, it should be on the sinks microservice.

Reply:
I also did restriction on sinks configuration thinking that will avoid peoples to use wrong headers in exporters request. The unique header that is used for prometheus like requests to control tenants. But I agree that have situations where we need other headers.